pkg/registry/core/namespace/storage/storage.go

```diff
@@ -150,6 +150,10 @@ func (r *REST) Delete(ctx context.Context, name string, options *metav1.DeleteOp
 	// upon first request to delete, we switch the phase to start namespace termination
 	// TODO: enhance graceful deletion's calls to DeleteStrategy to allow phase change and finalizer patterns
 	if namespace.DeletionTimestamp.IsZero() {
+		if err := deleteValidation(nsObj); err != nil {
```

Comment:
same comment here... call deleteValidation inside the GuaranteedUpdate callback with the actual object being deleted